File: Beakjoon/src/Silver/Lv4/boj_11399.java
package Silver.Lv4;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;
import java.util.stream.Stream;

public class boj_11399 { // ATM
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int[] people = Stream.of(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
        int result = 0;
        int temp = 0;
        Arrays.sort(people);

        for (int i = 0; i < N; i++) {
            temp += people[i];
            result += temp;
        }

        bw.write(result + "\n");
        bw.flush();
        bw.close();
    }
}

File: Beakjoon/src/Silver/Lv4/Boj_11399.kt
package Silver.Lv4

fun main() {
    val N: Int = readln().toInt()
    val people = readln().split(" ").map { it.toInt() }.toTypedArray()
    var temp = 0
    var result = 0

    people.sort()

    for (i in 0 until N) {
        temp += people[i]
        result += temp
    }

    println(result)
}
